pyccel/ast/headers.py

```diff
@@ -621,9 +621,9 @@ def __init__(self, name, args, master, master_args, results=None, results_shapes
         if not isinstance(name, str):
             raise TypeError('name must be of type str or PyccelSymbol')
 
-        # master can be a string or FunctionDef
-        if not isinstance(master, (str, FunctionDef, Interface)):
-            raise ValueError('Expecting a master name of FunctionDef')
+        # master can be a string or FunctionCall
+        if not isinstance(master, (str, FunctionCall)):
+            raise ValueError('Expecting a function name, or a FunctionCall')
```

Comment:
This must be how we ended up with `FunctionDefArgument`s, I thought this was just the name of the function used on the rhs, not the whole rhs